### ISSUE TITLE
npm run reference/test gets stuck

### ISSUE THREAD

**Issue by ashokkumar** [garris/BackstopJS#190]
While running the npm run reference/npm run test sometime it gets stuck. I did some debugging and found that in capture/genBitmaps.js on line number 88 this.thenOpen(url, function() the callback is getting called even after the page is loaded fully.

I tried setting the stepTimeout and not stopping the process onStepTimeout and continuing it fixed the issue.

But just trying to understand what I am doing wrong and why it is happening. Also would like to know anyone else complaint the same issue. Are other better solution to the problem.

My app is a ReactJS single page app.

**Comment by dixit4patel** [comment]
Can you give code sample here. I am also facing same issue. Its urgent.

**Comment by rbayliss** [comment]
Yeah, I'm experiencing a similar thing.  If it helps in tracing it back, here's a build that's getting hung during a Backstop test: https://circleci.com/gh/LastCallMedia/Drupal-Scaffold/62

And the code: https://github.com/LastCallMedia/Drupal-Scaffold/tree/c21dd658354179e4f86b5b114600a2e839e25f19/backstop
